server/src/weaverbird/backends/utils/formula_builder.py

```diff
@@ -0,0 +1,566 @@
+    # modulo from a division %
+    MODULO_OP: str = '%'
+    # float division /
+    DIVIDE_OP: str = '/'
+    # power operation ^
+    POW_OP: str = '^'
+
+    # The character that will be used to encapsulate columns
+    QUOTE_CHAR: str = ''
+
+    @classmethod
+    def clean_extras_spaces_and_tabs(cls, _str) -> str:
+        return ' '.join(_str.strip().split(' '))
+
+    @classmethod
+    def _node_as_constant(cls, constant: ast.Constant, columns_aliases: dict | None = None) -> Any:
+        return constant.value
+
+    @classmethod
+    def _node_as_name(cls, name: ast.Name, columns_aliases: dict | None = None) -> str:
+        """
+        Identifiers correspond to column names
+        """
+        return str(ColumnBuilder(cls.QUOTE_CHAR, name.id))
+
+    @classmethod
+    def _node_as_binop(cls, binop: ast.BinOp, columns_aliases: dict | None = None) -> Any:
+
+        binop_str = (
+            f" {cls._node_as_ast(binop.left, columns_aliases)}"
+            f" {cls.get_op(binop)}"
+            f" {cls._node_as_ast(binop.right, columns_aliases)}"
+        )
+
+        return binop_str
+
+    @classmethod
+    def _node_as_unaryop(cls, unop: ast.UnaryOp, columns_aliases: dict | None = None) -> Any:
+
+        unary_operator_str = {
+            'ast.USub': cls._node_as_ast,
+        }
+
+        try:
+            return unary_operator_str[str(type(unop.op))](unop.operand, columns_aliases)
+        except Exception as excp:
+            raise InvalidFormula(f'Operator {unop.op.__class__} is not supported') from excp
+
+    @classmethod
+    def _node_as_expr(
+        cls, expr: ast.Expr, columns_aliases: dict | None = None
+    ) -> dict[str, Any] | Any:
+
+        if isinstance(expr.value, ast.AST):
+            return cls._node_as_ast(expr.value, columns_aliases)
+        else:
+            raise InvalidFormula
+
+    @classmethod
+    def _node_as_ast(cls, node: ast.AST, columns_aliases: dict | None = None) -> Any:
+
+        formula_form = {
+            'BinOp': cls._node_as_binop,
+            'UnaryOp': cls._node_as_unaryop,
+            'Constant': cls._node_as_constant,
+            'Name': cls._node_as_name,
+        }
+
+        try:
+            formula_node_func = formula_form[type(node).__name__]
+            return formula_node_func(node, columns_aliases)  # type: ignore
+        except IndexError as excp:
+            raise InvalidFormula(f'Formula node {node} is not supported') from excp
+
+    @classmethod
+    def get_op(cls, binop: ast.BinOp) -> str:
+
+        operator_str = {
+            'Add': cls.ADD_OP,
+            'Sub': cls.SUBSTRACT_OP,
+            'Mult': cls.MULTIPLY_OP,
+            'Div': cls.DIVIDE_OP,
+            'Pow': cls.POW_OP,
+        }
+
+        try:
+            return operator_str[type(binop.op).__name__]
+        except IndexError as excp:
+            raise InvalidFormula(f'Operator {binop.op.__class__} is not supported') from excp
+
+    @classmethod
+    def evaluate(cls, formula: Any):
```

Comment:
Do we need this here? Each backend will use its own engine to evaluate the formula, so I guess this formula_builder could stick to just proving the syntax tree.